activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -53,6 +53,16 @@ def not(opts, *rest)
         @scope.where_clause += where_clause.invert
         @scope
       end
+
+      def missing(*args)
+        args.each do |arg|
+          opts = { arg => { id: nil } }
```

Comment:
I think we shouldn't use a fixed column name like `id`. Because Rails also supports customizing primary keys for tables. So you should try to figure out the column name that the assocition uses as primary key

Reply:
@st0012 great suggestion! I've incorporated this into my latest commit.